test/e2e/testcase/testcase.go

```diff
@@ -98,21 +101,418 @@ func RunTestCases(f *framework.Framework) {
 			gomega.Expect(err).To(gomega.BeNil())
 		})
 
-		ginkgo.It("service qualities", func() {
+		ginkgo.Describe("service qualities with opsState and metadata", func() {
 
-			// deploy
-			gss, err := f.DeployGssWithServiceQualities()
-			gomega.Expect(err).To(gomega.BeNil())
+			ginkgo.BeforeEach(func() {
+				f.MarkTestStart()
+			})
 
-			err = f.ExpectGssCorrect(gss, []int{0, 1, 2})
-			gomega.Expect(err).To(gomega.BeNil())
+			ginkgo.AfterEach(func() {
+				gomega.Expect(f.AfterEach()).Should(gomega.BeNil())
+				if ginkgo.CurrentGinkgoTestDescription().Failed {
+					f.DumpAuditIfFailed()
+				}
+			})
 
-			err = f.WaitForGsUpdatePriorityUpdated(gss.GetName()+"-0", "20")
-			gomega.Expect(err).To(gomega.BeNil())
-			err = f.WaitForGsUpdatePriorityUpdated(gss.GetName()+"-1", "20")
-			gomega.Expect(err).To(gomega.BeNil())
-			err = f.WaitForGsUpdatePriorityUpdated(gss.GetName()+"-2", "20")
-			gomega.Expect(err).To(gomega.BeNil())
+			ginkgo.It("service qualities update opsState and metadata", func() {
+				// deploy basic gss
+				gss, err := f.DeployGameServerSet()
+				gomega.Expect(err).To(gomega.BeNil())
+
+				err = f.WaitForGsCreated(gss)
+				gomega.Expect(err).To(gomega.BeNil())
+
+				// Wait for all GameServers to be fully Ready before patching serviceQualities
+				gomega.Eventually(func() bool {
+					gs0, err := f.GetGameServer(gss.GetName() + "-0")
+					if err != nil {
+						return false
+					}
+					gs1, err := f.GetGameServer(gss.GetName() + "-1")
+					if err != nil {
+						return false
+					}
+					gs2, err := f.GetGameServer(gss.GetName() + "-2")
+					if err != nil {
+						return false
+					}
+					// Ensure all have Ready state and Running phase
+					return gs0.Status.CurrentState == "Ready" &&
+						gs1.Status.CurrentState == "Ready" &&
+						gs2.Status.CurrentState == "Ready"
+				}, "60s", "2s").Should(gomega.BeTrue())
+
+				// patch serviceQualities with opsState and metadata actions
+				up := intstr.FromInt(20)
+				dp := intstr.FromInt(10)
+				serviceQualities := []gameKruiseV1alpha1.ServiceQuality{
+					{
+						Name:          "health-check",
+						ContainerName: "game-server",
```

Comment:
It doesn't match the container name deployed by default. https://github.com/sonalgaud12/kruise-game/blob/40acb3154c43865ae8e6ac17205d09edd401eeb3/test/e2e/client/client.go#L22-L26